python/lsst/ts/m2/mock_server.py

```diff
@@ -0,0 +1,924 @@
+        Can be set to `socket.AF_INET` or `socket.AF_INET6` to limit the server
+        to IPv4 or IPv6, respectively. If `socket.AF_UNSPEC` (the default)
+        the family will be determined from host, and if host is None,
+        the server may listen on both IPv4 and IPv6 sockets.
+
+    Attributes
+    ----------
+    log : `logging.Logger`
+        A logger.
+    server_cmd : `tcpip.OneClientServer`
+        Command server.
+    server_tel : `tcpip.OneClientServer`
+        Telemetry server.
+    """
+
+    def __init__(self, host, port_cmd, port_tel, log=None, family=socket.AF_UNSPEC):
+
+        # Set the logger
+        if log is None:
+            self.log = logging.getLogger(type(self).__name__)
+        else:
+            self.log = log.getChild(type(self).__name__)
+
+        self.server_cmd = tcpip.OneClientServer(
+            "Commands",
+            host,
+            port_cmd,
+            self.log,
+            self._connect_callback_cmd,
+            family=family,
+        )
+        self.server_tel = tcpip.OneClientServer(
+            "Telemetry",
+            host,
+            port_tel,
+            self.log,
+            self._connect_callback_tel,
+            family=family,
+        )
+
+        # Monitor loop task of command server (asyncio.Future)
+        self._monitor_loop_task_cmd = salobj.make_done_future()
+
+        # Monitor loop task of telemetry server (asyncio.Future)
+        self._monitor_loop_task_tel = salobj.make_done_future()
+
+        # Send the one-time message
+        self._send_one_time_msg_done = False
+
+        # System is in the Enabled state or not
+        self._is_enabled = False
+
+    def _connect_callback_cmd(self, server_cmd):
+        """Called when the command server connection state changes.
+
+        Parameters
+        ----------
+        server_cmd : `tcpip.OneClientServer`
+            Command server.
+        """
+        self._monitor_loop_task_cmd.cancel()
+
+        if self.server_cmd.connected:
+            self._monitor_loop_task_cmd = asyncio.create_task(self._monitor_msg_cmd())
+
+    async def _monitor_msg_cmd(self, period_check=0.001):
+        """Monitor the message of incoming command.
+
+        Parameters
+        ----------
+        period_check : `float`, optional
+            Period of checking the incoming messages in second. (the default is
+            0.001)
+        """
+
+        try:
+            while True:
+
+                if not self._send_one_time_msg_done:
+                    await self._send_one_time_msg()
+                    self._send_one_time_msg_done = True
+
+                if self.server_cmd.reader.at_eof():
+                    self.log.info("Command reader at eof; closing client")
+                    break
+
+                await self._process_msg_cmd(period_check)
+
+        except ConnectionError:
+            self.log.info("Command reader disconnected; closing client")
+            self._monitor_loop_task_cmd.cancel()
```

Comment:
After the `ConnectionError`, it is better to cancel the task to avoid the resource problem (`iPython` might complain this sometimes). I did not do this originally, and after the multiple tests, I realized that this step might be needed. Thanks!